Zend/zend_compile.c

```diff
@@ -6254,6 +6254,17 @@ zend_op *zend_compile_class_decl(zend_ast *ast, zend_bool toplevel) /* {{{ */
 		}
 	}
 
+	if (ce->__set) {
+		if (ce->__set->common.fn_flags & ZEND_ACC_STATIC) {
+			zend_error_noreturn(E_COMPILE_ERROR, "Setter %s::%s() cannot be static",
+				ZSTR_VAL(ce->name), ZSTR_VAL(ce->__set->common.function_name));
+		} else if (ce->__set->common.fn_flags & ZEND_ACC_HAS_RETURN_TYPE) {
+			zend_error_noreturn(E_COMPILE_ERROR,
```

Comment:
This is quite serious BC break: https://3v4l.org/8kqva

Reply:
Yeap, it's, but is to ensure the return type here, avoiding scenarios like: https://3v4l.org/Yd3le.

Also, I do plan to discuss this on internals, but I want first discuss the implementation itself :)

Reply:
The code you referenced is just fine: https://3v4l.org/hP5T7

Reply:
@Majkl578 In your code you are using `__set` as a method itself, not as a magic one, right? So this is your point of the BC break?

With this PR, I'm trying to cover the magic one: https://3v4l.org/0nUDB.

Reply:
Welp, magic method is still a regular method, only also used for some _magic_. :)